### PR TITLE
feat: deployment executables

### PR DESCRIPTION
**Pull request by bashleigh**
General deployment executable function actions

### CODE REVIEW

File: packages/deployment-service/src/executables/pull.ts [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
+  zip.extractAllTo(`${dir}`, true)
+}
+
+export const pull: ExecutableType = async (task: TaskEntity, pipeline: PipelineEntity): Promise<true | never> => {
   console.log('pull...')
   console.log('executable', task)
 
+  try {
+    await new Promise<void>((resolve) => {
+      request(`https://codeload.github.com/${pipeline.repository}/zip/refs/heads/master`)
```
Comment (willmcvay) [inline packages/deployment-service/src/executables/pull.ts:19]: Be nice if this were a constant - the base uri that is

File: packages/deployment-service/src/executables/pull.ts [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+        .pipe(fs.createWriteStream(`${dir}/${cloneZip}`))
+        .on('close', function () {
+          resolve()
+        })
+    })
+
+    // console.log(result)
```
Comment (willmcvay) [inline packages/deployment-service/src/executables/pull.ts:26]: Couple of comments

**Review by willmcvay — APPROVED** [review]
Just a couple of small bits

File: packages/deployment-service/src/executables/pull.ts (hunk 2 above)
Comment (bashleigh) [reply packages/deployment-service/src/executables/pull.ts:26]: this'll all be updated and removed :)

**Pull request merged**